code/modules/antagonists/heretic/knowledge/side_rust_cosmos.dm

```diff
@@ -18,6 +18,26 @@
 	cost = 1
 	route = PATH_SIDE
 
+/datum/heretic_knowledge/entropy_pulse
+	name = "Pulse of Entropy"
+	desc = "Allows you to transmute 20 irons and 2 garbage items to fill the surrounding vicinity of the rune with rust."
+	gain_text = "Reality begins to whisper to me. To give it its entropic end."
+	required_atoms = list(
+		/obj/item/stack/sheet/iron = 20,
+		/obj/item/trash = 2
+	)
+	cost = 1
+	route = PATH_SIDE
+	var/rusting_range = 4
+
+/datum/heretic_knowledge/entropy_pulse/on_finished_recipe(mob/living/user, list/selected_atoms, turf/loc)
+	var/turf/ritual_turf = loc
+	for(var/turf/nearby_turf in view(rusting_range, ritual_turf))
+		if(prob(20) || istype(nearby_turf, /turf/closed))
```

Comment:
```suggestion
	for(var/turf/nearby_turf in view(rusting_range, ritual_turf))
		// Closed turfs purposefully excluded to avoid revealing your hideout instantly 
		if(prob(20) || isclosedturf(nearby_turf))
```
- Use of ishelpers
- Might as well leave a comment to explain why it does what it does.